python/kserve/kserve/protocol/rest/server.py

```diff
@@ -26,19 +26,32 @@
 from timing_asgi import TimingClient, TimingMiddleware
 from timing_asgi.integrations import StarletteScopeToName
 
-from kserve.errors import (InferenceError, InvalidInput, ModelNotFound,
-                           ModelNotReady, generic_exception_handler,
-                           inference_error_handler, invalid_input_handler,
-                           model_not_found_handler, model_not_ready_handler,
-                           not_implemented_error_handler)
+from kserve.errors import (
+    InferenceError,
+    InvalidInput,
+    ModelNotFound,
+    ModelNotReady,
+    generic_exception_handler,
+    inference_error_handler,
+    invalid_input_handler,
+    model_not_found_handler,
+    model_not_ready_handler,
+    not_implemented_error_handler,
+)
 from kserve.logging import trace_logger
 from kserve.protocol.dataplane import DataPlane
 
+from .openai.config import maybe_register_openai_endpoints
```

Comment:
We can do try/catch import to make openai dependency optional

Reply:
it is still beneficial to make this optional as we have model servers that do not implement openai endpoints

Reply:
I don't think this try/catch is needed as the `.openai.config` module guards all loading of `openai` already

Reply:
seems like this line did not work
https://github.com/kserve/kserve/pull/3477/files#diff-772e23576926728b2bb70421428622fe6163a87d22b616ad264b168895d11501R27

Reply:
Turns out `find_spec()` doesn't return any error if module is not found, so `openai_is_available()` is always `True`. Fixed this with https://github.com/kserve/kserve/pull/3477/commits/aecd2cea8e8c1d53177cfb571260c5a7a760cdf1.